### PR TITLE
chore(breadcrumbs): Adds stories for navigation

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

File: packages/clay-navigation/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import React from 'react';
+import {Breadcrumb} from './Breadcrumb';
+
+const ClayNavigation: React.FunctionComponent<
+	React.HTMLAttributes<HTMLDivElement>
+> & {
+	Breadcrumb: typeof Breadcrumb;
+} = ({className, ...otherProps}) => {
+	return (
+		<div {...otherProps} className={classNames(className)}>
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/index.tsx]: Is this just essentially a placeholder for now?

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import classNames from 'classnames';
+import React from 'react';
+
+export interface IBreadcrumbItem {
+	active?: boolean;
+
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]: no need for line breaks here unless you are adding jsdocs

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import React, {useState} from 'react';
+import {IBreadcrumbItem} from './BreadcrumbItem';
+
+export interface IBreadcrumbEllipsisProps {
+	items?: Array<IBreadcrumbItem>;
+
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: no need for line breaks here unless you are adding jsdocs

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+import warning from 'warning';
+import {BreadcrumbEllipsis} from './BreadcrumbEllipsis';
+import {BreadcrumbItem, IBreadcrumbItem} from './BreadcrumbItem';
+
+interface IProps extends React.HTMLAttributes<HTMLOListElement> {
+	ellipsisBuffer?: number;
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: We will need JSDocs for these props since this is the public facing API

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+	ellipsisProps?: Object;
+
+	items: Array<IBreadcrumbItem>;
+
+	spritemap: string;
+}
+
+const getBufferList = (
+	items: Array<IBreadcrumbItem>,
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: Generally I think we tend to opt for `IBreadcrumbItem[]` for array types. I may be wrong, or we might need to add an eslint rule for this.

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+	start: number,
+	end: number,
+	ellipsisProps: Object,
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: instead of an explicit prop here, can we just use `...otherProps` since bufferList implies it is for an ellipsis

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+		: removedItems;
+};
+
+const getBreadcrumbItems = (
+	ellipsisBuffer: number,
+	items: Array<IBreadcrumbItem>,
+	ellipsisProps: Object,
+	spritemap: string
+) => {
+	const activeIndex = items.findIndex(item => item.active);
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: Im not sure this is supported in IE11. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#Browser_compatibility

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	...otherProps
+}: IProps) => {
+	warning(
+		items.findIndex(item => item.active) != -1,
+		'ClayNavigation.Breadcrumb expects at least one `active` item on `items`.'
+	);
+
+	return (
+		<ol {...otherProps} className={classNames('breadcrumb', className)}>
+			{(ellipsisBuffer
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: could we move this logic outside of the return just to be a little more clear? We could set it to `breadCrumbItems` and then just only have to use `breadCrumbItems.map(`. I think this will help readability a bit.

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	spritemap: string;
+}
+
+export const BreadcrumbEllipsis: React.FunctionComponent<
+	IBreadcrumbEllipsisProps
+> = ({items, spritemap, ...otherProps}) => {
+	const [active, setActive] = useState(false);
+	return (
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: line break above this. Essentially we put line breaks where "order is important", so if we move the return above the `const`, it'd break.

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+						items,
+						ellipsisProps,
+						spritemap
+				  )
+				: items
+			).map((item: IBreadcrumbItem | React.ReactElement, i: number) =>
+				React.isValidElement(item) ? (
+					React.cloneElement(item, {key: `ellipsis${i}`})
+				) : (
+					<BreadcrumbItem
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: What about `onClick` instead of href? I could imagine that would be used as well.

File: packages/clay-navigation/src/__tests__/index.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+			<ClayNavigation.Breadcrumb
+				ellipsisBuffer={1}
+				ellipsisProps={{style: {fontSize: '15px'}}}
+				items={items}
+				spritemap="path/to/spritemap"
+			/>
+		);
+
+		expect(container).toMatchSnapshot();
+	});
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/__tests__/index.tsx]: Can we add some more tests for interactions, such as dropdown interactions. Also, if we add `onClick` functionality, we could test that interaction as well.

File: packages/clay-navigation/src/index.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import classNames from 'classnames';
+import React from 'react';
+import {Breadcrumb} from './Breadcrumb';
+
+const ClayNavigation: React.FunctionComponent<
+	React.HTMLAttributes<HTMLDivElement>
+> & {
+	Breadcrumb: typeof Breadcrumb;
+} = ({className, ...otherProps}) => {
+	return (
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/index.tsx]: I would do the same thing we are doing for `ClayForm`, the content of this is not necessary.

File: packages/clay-navigation/stories/index.tsx [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+				},
+				{
+					href: '#11',
+					label: 'Eleven',
+				},
+			]}
+			spritemap={spritemap}
+		/>
+	))
+	.add('should throw a warning', () => (
```
Comment (matuzalemsteles) [inline packages/clay-navigation/stories/index.tsx]: I didn't quite understand why.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 4 above)
Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Just one comment would not be enough?

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 5 above)
Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: I think they are just different ways of doing this, I like `Array<...>` because it is more explicit to me and goes along with the other syntaxes like `React.FunctionComponent<...>`.

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+					items={removedItems}
+					key="breadcrumbEllipsis"
+					spritemap={spritemap}
+					{...ellipsisProps}
+				/>,
+		  ]
+		: removedItems;
+};
+
+const getBreadcrumbItems = (
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/Breadcrumb.tsx]: I see this and `getBufferList` is akin to the pagination rule, so I would choose to try to use the same function for the two and add some new params to customize some components... I think it would be best not to have rules duplicated in different places.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 10 above)
Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: I think we can keep both. But we have to think if keeping `href` is probably for cases that would have been a navigation using the browser and not a navigation through the React Router for example. One thought about this is:

- If you allow `href`, it may be cases where people will only use the component as if it were a taglib. Maybe I was wrong, but in v2 we added the href for those cases.
- If you allow `onClick`, it may be cases where you have control and want to do a navigation using the React Router for example.

So the question is, can we just allow `onClick` and force people to use components for full React applications rather than for a loose component, such as a taglib. But if you follow this path it can not be used in a taglib.

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+			trigger={
+				<ClayButton className="breadcrumb-link" displayType="unstyled">
+					<ClayIcon spritemap={spritemap} symbol="ellipsis-h" />
+					<ClayIcon spritemap={spritemap} symbol="caret-bottom" />
+				</ClayButton>
+			}
+			{...otherProps}
+		>
+			<ClayDropDown.ItemList>
+				{items &&
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: I'd rather leave the default value of items an empty array and so we do not need the condition here.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 5 above)
Comment (bryceosterhaus) [reply packages/clay-navigation/src/Breadcrumb.tsx]: @matuzalemsteles yeah thats fine, we likely should just pick one or the other and stick with it for consistency

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 10 above)
Comment (bryceosterhaus) [reply packages/clay-navigation/src/Breadcrumb.tsx]: For a react app with routing, I think the user would also just use an `href` and then let the touer handle changing the state. The more I think about it, the more `onClick` is likely used less and maybe wouldn't be used at all.

Comment (bryceosterhaus) [reply packages/clay-navigation/src/Breadcrumb.tsx]: So I guess im actually thinking the opposite, and only allowing href and keep them as anchor tags

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 5 above)
Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Sure, I think we can only use `Array<...>` for consistency, is there any way to add that in lint?

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 10 above)
Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Oh no 😅, I'm not sure if `href` would help with the React Router because you need to use the `Link` component (at least that's how I used it 🙂).

So I would probably use `onClick` and then call `navigate` (I think this is Reach Router 🙂).

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 5 above)
Comment (bryceosterhaus) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Yeah I think we can add, https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/array-type.md#generic

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 10 above)
Comment (bryceosterhaus) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Oh good point with `Link`. Maybe we need to add ability to pass in a custom component as well, since that is likely a important use case.

The reason I would want to avoid `onClick` only is because you wouldn't be able to right click open or hold cmd and click to open in a new tab. href is important for accessibility.

Comment (matuzalemsteles) [reply packages/clay-navigation/src/Breadcrumb.tsx]: > Oh good point with Link. Maybe we need to add ability to pass in a custom component as well, since that is likely a important use case.

Hmm, I had not thought about it, but I find it interesting.

> The reason I would want to avoid onClick only is because you wouldn't be able to right click open or hold cmd and click to open in a new tab. href is important for accessibility.

Yes you are right, we can keep both, I think both end up being important.

File: packages/clay-navigation/src/index.tsx (hunk 1 above)
Comment (diegonvs) [reply packages/clay-navigation/src/index.tsx]: Nope, I miss it.

File: packages/clay-navigation/src/BreadcrumbItem.tsx (hunk 2 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbItem.tsx]: Solved!

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx (hunk 3 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: Solved!

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 4 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Added 
```javascript
/**
 * The number of Breadcrumb Items to show on each side of the active Breadcrumb Item before
 * using an ellipsis dropdown.
 */
```

It's okay?

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 5 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: I vote for Array<anything> too :)

Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: https://github.com/liferay/clay/pull/2169

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 6 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: `otherProps` is already being used on `<ol>` root

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 7 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Replacing findIndex with a filter length.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 8 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Makes sense :)

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx (hunk 9 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: I didn't know that. Thanks for the tip!

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 10 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Was added a property called `onItemClicked`

File: packages/clay-navigation/src/__tests__/index.tsx (hunk 11 above)
Comment (diegonvs) [reply packages/clay-navigation/src/__tests__/index.tsx]: For sure

File: packages/clay-navigation/stories/index.tsx (hunk 13 above)
Comment (diegonvs) [reply packages/clay-navigation/stories/index.tsx]: Removed.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 14 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Done!

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx (hunk 15 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: Done!

File: packages/clay-pagination/src/index.tsx [hunk 16]
```diff
@@ -157,7 +105,7 @@ const ClayPagination: React.FunctionComponent<IProps> = ({
 						href={
 							hrefConstructor && hrefConstructor(page as number)
 						}
-						key={page as number}
+						key={`page${index}` as string}
```
Comment (matuzalemsteles) [inline packages/clay-pagination/src/index.tsx]: I think it would be better for the key to be `page`. It may happen that `page${index}` change when any item becomes a DropDown, page does not.

File: packages/clay-pagination/src/__tests__/index.tsx [hunk 17]
```diff
@@ -118,7 +118,10 @@ describe('ClayPagination', () => {
 
 		fireEvent.click(getByText(container, '...') as HTMLElement, {});
 
-		fireEvent.click(getByText(document.body, '4') as HTMLElement, {});
+		fireEvent.click(
```
Comment (matuzalemsteles) [inline packages/clay-pagination/src/__tests__/index.tsx]: Out of curiosity, why did you change that?

Comment (diegonvs) [reply packages/clay-pagination/src/__tests__/index.tsx]: Source formatting

Comment (diegonvs) [reply packages/clay-pagination/src/__tests__/index.tsx]: Source formatting

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+	 * Use this property for defining `otherProps` that will be passed to ellipsis dropdown trigger.
+	 */
+	ellipsisProps?: Object;
+
+	/**
+	 * Property to define Breadcrumb's items.
+	 */
+	items: Array<IBreadcrumbItem>;
+
+	spritemap: string;
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: For `spritemap`? In some components, we left out JSDoc of `spritemap`

File: packages/clay-pagination/src/index.tsx (hunk 16 above)
Comment (diegonvs) [reply packages/clay-pagination/src/index.tsx]: What do you think about `paginationItem${index}`?

Comment (diegonvs) [reply packages/clay-pagination/src/index.tsx]: Solved internally :)

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+					data-testid={`testId${label}`}
+					href={href}
+					role="button"
+				>
+					{label}
+				</a>
+			);
+		}
+
+		if (onItemClick) {
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]: I think we can do the same here that was done in `href` just render the `ClayButton` if the item is not active.

File: packages/clay-navigation/src/Breadcrumb.tsx [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+				BreadcrumbEllipsis,
+				ellipsisProps,
+				items.indexOf(activeItems[0]),
+				spritemap
+		  )
+		: items;
+
+	return (
+		<ol {...otherProps} className={classNames('breadcrumb', className)}>
+			{breadCrumbItems.map((item: any, i: number) =>
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/Breadcrumb.tsx]: Can we fix this `any` type here?

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+import ClayDropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import React, {useState} from 'react';
+import {IBreadcrumbItem} from './BreadcrumbItem';
+
+export interface IBreadcrumbEllipsisProps {
+	/**
+	 * Property to define BreadcrumbEllipsis's items.
+	 */
+	items?: Array<IBreadcrumbItem>;
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: Is this really an optional param? Meaning, would we ever expect or want this component not to receive items?

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: This should be optional since spritemap can come via context

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+
+	/**
+	 * Label of the Breadcrumb item
+	 */
+	label: string;
+
+	/**
+	 * Callback for when a Breadcrumb item is clicked.
+	 */
+	onItemClick?: (event: React.SyntheticEvent) => void;
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]: Since its already called a `BreadcrumbItem`, can we just call this `onClick` to reduce name redundancy?

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+export const BreadcrumbItem = ({
+	active,
+	href,
+	label,
+	onItemClick,
+	...otherProps
+}: IBreadcrumbItem) => {
+	const ItemElement = ({
```
Comment (bryceosterhaus) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]: How come this is defined separately as its own function? We are only calling this once, which means we can likely just move this to the jsx markup

File: packages/clay-shared/src/getEllipsisItems.tsx [hunk 25]
```diff
@@ -0,0 +1,79 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import React from 'react';
+
+const getBufferList = (
```
Comment (bryceosterhaus) [inline packages/clay-shared/src/getEllipsisItems.tsx]: Instead of having a ton of arguments, lets just make this accept a config object. Usually is there are more than 2 or 3 args, its easier to pass in an object since its more explicit.

File: packages/clay-shared/src/getEllipsisItems.tsx [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+	start: number,
+	end: number,
+	EllipsisComponent: React.ComponentType<any>,
+	ellipsisProps?: Object,
+	spritemap?: string
+): Array<Object | number | React.ReactElement> => {
+	const removedItems = items.slice(start, Math.max(end, start));
+
+	if (removedItems.length > 1) {
+		return [
```
Comment (bryceosterhaus) [inline packages/clay-shared/src/getEllipsisItems.tsx]: How come we are returning an array here?

File: packages/clay-shared/src/getEllipsisItems.tsx [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+				{...ellipsisProps}
+			/>,
+		];
+	}
+
+	return removedItems;
+};
+
+export const getEllipsisItems = (
+	ellipsisBuffer: number,
```
Comment (bryceosterhaus) [inline packages/clay-shared/src/getEllipsisItems.tsx]: Same as above, lets move all these args into a config object.

File: packages/clay-navigation/src/Breadcrumb.tsx (hunk 20 above)
Comment (diegonvs) [reply packages/clay-navigation/src/Breadcrumb.tsx]: Fixed!

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx (hunk 21 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: I left this optional due I was initializing with an empty array and I was taking as a base PaginationEllipsis that this property is defined as optional. Perhaps, make sense turning this one obligatory.

File: packages/clay-navigation/src/BreadcrumbEllipsis.tsx (hunk 22 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbEllipsis.tsx]: Makes sense. I'm currently updating all components that receive `spritemap` to be optional and update docs.

File: packages/clay-navigation/src/BreadcrumbItem.tsx (hunk 24 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbItem.tsx]: Refactoring, I don't remember why I did that but possibly this problem was solved.

File: packages/clay-navigation/src/BreadcrumbItem.tsx (hunk 23 above)
Comment (diegonvs) [reply packages/clay-navigation/src/BreadcrumbItem.tsx]: Yes, makes sense.

File: packages/clay-shared/src/getEllipsisItems.tsx (hunk 25 above)
Comment (diegonvs) [reply packages/clay-shared/src/getEllipsisItems.tsx]: Done!

File: packages/clay-shared/src/getEllipsisItems.tsx (hunk 26 above)
Comment (diegonvs) [reply packages/clay-shared/src/getEllipsisItems.tsx]: Since the return must be an array, makes sense encapsulate the component in an array previously.

File: packages/clay-shared/src/getEllipsisItems.tsx (hunk 27 above)
Comment (diegonvs) [reply packages/clay-shared/src/getEllipsisItems.tsx]: Done!

File: packages/clay-navigation/stories/index.tsx [hunk 28]
```diff
@@ -0,0 +1,125 @@
...
+import '@clayui/css/lib/css/atlas.css';
+import ClayNavigation from '../src';
+import React from 'react';
+import {number} from '@storybook/addon-knobs';
+import {storiesOf} from '@storybook/react';
+
+const spritemap = require('@clayui/css/lib/images/icons/icons.svg');
+
+storiesOf('ClayNavigation', module)
+	.add('default', () => (
```
Comment (matuzalemsteles) [inline packages/clay-navigation/stories/index.tsx]: You can rename it to `/w breadcrumb`? Since we can have several other components here.

File: packages/clay-navigation/stories/index.tsx [hunk 29]
```diff
@@ -0,0 +1,125 @@
...
+				},
+				{
+					href: '#11',
+					label: 'Eleven',
+				},
+			]}
+			spritemap={spritemap}
+		/>
+	))
+	.add('without anchors', () => {
```
Comment (matuzalemsteles) [inline packages/clay-navigation/stories/index.tsx]:
```suggestion
	.add('/w breadcrumb without anchors', () => {
```

File: packages/clay-navigation/stories/index.tsx [hunk 30]
```diff
@@ -0,0 +1,125 @@
...
+
+import '@clayui/css/lib/css/atlas.css';
+import ClayNavigation from '../src';
+import React from 'react';
+import {number} from '@storybook/addon-knobs';
+import {storiesOf} from '@storybook/react';
+
+const spritemap = require('@clayui/css/lib/images/icons/icons.svg');
+
+storiesOf('ClayNavigation', module)
```
Comment (matuzalemsteles) [inline packages/clay-navigation/stories/index.tsx]: I would be curious to know how `Breadcrumbs` works when controlled with `onClick`, we may have an example of usage just like pagination.

File: packages/clay-navigation/stories/index.tsx (hunk 28 above)
Comment (diegonvs) [reply packages/clay-navigation/stories/index.tsx]: makes sense. Maybe in the future, we can create a folder here.

File: packages/clay-navigation/stories/index.tsx (hunk 30 above)
Comment (diegonvs) [reply packages/clay-navigation/stories/index.tsx]: Sent!

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 31]
```diff
@@ -0,0 +1,74 @@
...
+			<a
+				className="breadcrumb-link"
+				data-testid={`testId${label}`}
+				href={href}
+				role="button"
+			>
+				{label}
+			</a>
+		) : onClick ? (
+			<ClayButton
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]: I was observing one thing when I see the buttons the mark up should be as follows:

```jsx
<ClayButton
	className="breadcrumb-link"
	data-testid={`testId${label}`}
	displayType="unstyled"
	onClick={onClick}
	title={label}
>
	<span className="breadcrumb-text-truncate">
		{label}
	</span>
</ClayButton>
```

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 32]
```diff
@@ -0,0 +1,74 @@
...
+		{!active && href ? (
+			<a
+				className="breadcrumb-link"
+				data-testid={`testId${label}`}
+				href={href}
+				role="button"
+			>
+				{label}
+			</a>
+		) : onClick ? (
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]:
```suggestion
		) : !active && onClick ? (
```

File: packages/clay-navigation/src/BreadcrumbItem.tsx [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+		{...otherProps}
+	>
+		{!active && href ? (
+			<a
+				className="breadcrumb-link"
+				data-testid={`testId${label}`}
+				href={href}
+				role="button"
+			>
+				{label}
```
Comment (matuzalemsteles) [inline packages/clay-navigation/src/BreadcrumbItem.tsx]:
```jsx
<span className="breadcrumb-text-truncate">
    {label}
</span>
```
